src/EditorFeatures/CSharpTest/ConvertToInterpolatedString/ConvertPlaceholderToInterpolatedStringTests.cs

```diff
@@ -19,27 +21,131 @@ public class ConvertPlaceholderToInterpolatedStringTests : AbstractCSharpCodeAct
         protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
             => new CSharpConvertPlaceholderToInterpolatedStringRefactoringProvider();
 
-        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertToInterpolatedString)]
-        public async Task TestSingleItemSubstitution()
+        public static IEnumerable<object[]> InvocationData
+        {
+            get
+            {
+                // Every API so far starts to use a params object after 4 paramters following the formatted 
+                const int ParametersToCheck = 4;
+
+                // string.Format gets replaced with just the interpolated string
+                for (var i = 1; i <= ParametersToCheck; i++)
+                {
+                    var invocation = $"string.Format({MakeFormattedParameters(i)})";
+                    var result = $"${MakeInterpolatedString(i)}";
+                    yield return new[] { invocation, result };
+                }
+
+                // These functions remain, but will change to take the
+                // interpolated string rather than call the string format 
+                var functionsToCheck = new[]
+                {
+                    "Console.Write",
+                    "Console.WriteLine",
+                    "Debug.WriteLine",
+                    "Debug.Print",
+                    "Trace.TraceError",
+                    "Trace.TraceWarning",
+                    "Trace.TraceInformation",
+                    //"TraceSource.TraceInformation" TODO: Trace source needs NetCoreApp, which tests don't run in. 
+                };
+
+                for (var i = 1; i <= ParametersToCheck; i++)
+                {
+                    foreach (var function in functionsToCheck)
+                    {
+                        var invocation = $"{function}({MakeFormattedParameters(i)})";
+                        var result = $"{function}(${MakeInterpolatedString(i)})";
+                        yield return new[] { invocation, result };
+                    }
+                }
+
+                // Makes a string of form "{0} {1}..."
+                static string MakeInterpolatedString(int numberOfParameters)
+                {
+                    var interpolatedString = "\"";
+
+                    for (var i = 0; i < numberOfParameters; i++)
+                    {
+                        interpolatedString += $"{{{i}}}";
```

Comment:
I cannot enumerate the ways in which I don't like this, but I'm going to grit my teeth and ignore it :D

Reply:
I'm not in love with this whole thing, but it seemed the best way to write it... What would you suggest? I''m happy to at least take it and ignore it :P

Reply:
oh yeah, the whole thing is annoying because its test code that is bordering on being more complex than the real code. I was just having a whinge about the interpolation here, so big deal.